fabric-api-base/src/testmodClient/java/net/fabricmc/fabric/test/base/client/FabricApiAutoTestClient.java

```diff
@@ -136,18 +129,19 @@ private void runTest() {
 			takeScreenshot("game_menu");
 			clickScreenButton("menu.returnToMenu");
 			waitForScreen(TitleScreen.class);
+			waitForServerStop();
 		}
 
 		try (var server = new TestDedicatedServer()) {
 			connectToServer(server);
 			waitForWorldTicks(5);
 
-			final GameProfile profile = submitAndWait(MinecraftClient::getGameProfile);
+			final GameProfile profile = computeOnClient(() -> MinecraftClient.getInstance().getGameProfile());
```

Comment:
Is there a specific reason you decided not to pass the client though as a param to the lambda? What you have here isnt wrong, but my worry is this makes it possibly easier to call `MinecraftClient.getInstance()` on the wrong thread.

An option might be to just check the thread in `getInstance()`, but not something for now, at least until this becomes a public API. Ill take a proper look at the changes soon.

Reply:
Honestly the only reason was that I didn't want to make more types of throwing functional interfaces. But it would definitely be convenient to have a `MinecraftClient` passed as the argument

Reply:
I was looking for something like this, thanks!

Reply:
> An option might be to just check the thread in `getInstance()`, but not something for now, at least until this becomes a public API. Ill take a proper look at the changes soon.

I wanted to do this but I thought it would be too controversial for this PR to be accepted, so I left the discussion for later. I can do it now if you want.